Code/Source/Utilities/ROS2Names.cpp

```diff
@@ -25,8 +27,92 @@ namespace ROS2
+            // Also, starting with '_' is not desired unless explicit. Topics/namespaces/parameters starting with "_" are hidden by default.
+            const AZStd::string prependToNumberStart = "o3de_";
+            rosified = prependToNumberStart + rosified;
+        }
+
+        if (input != rosified)
+        {
+            AZ_TracePrintf("RosifyName", "Name '%s' has been changed to '%s' to conform with ros2 naming restrictions", input.c_str(), rosified.c_str());
+        }
+        return rosified;
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateNamespace(const AZStd::string& ros2Namespace)
+    {
+        auto ros2GlobalizedNamespace = ros2Namespace;
+        const char namespacePrefix = '/';
+        if (ros2Namespace.find(namespacePrefix, 0) != 0)
+        {  // Prepend "/" if not included, this is done automatically by rclcpp so "/"-less namespaces are ok.
+            ros2GlobalizedNamespace = namespacePrefix + ros2Namespace;
+        }
+
+        int validationResult = 0;
+        auto ret = rmw_validate_namespace(ros2GlobalizedNamespace.c_str(), &validationResult, NULL);
+        if (ret != RMW_RET_OK)
+        {
+            AZ_Error("ValidateNamespace", false, "Call to rmw validation for namespace failed");
+            return AZ::Failure(AZStd::string("Unable to validate namespace due to rmw error"));
+        }
+
+        if (validationResult != RMW_NAMESPACE_VALID)
+        {
+            return AZ::Failure(AZStd::string(rmw_namespace_validation_result_string(validationResult)));
+        }
+
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateNamespaceField(void* newValue, [[maybe_unused]] const AZ::Uuid& valueType)
+    {
+        AZStd::string ros2Namespace(static_cast<const char*>(newValue));
+        return ValidateNamespace(ros2Namespace);
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateTopic(const AZStd::string& topic)
+    {
+        int validationResult = 0;
+        size_t invalidIndex; // Unused. We could choose to display it in information, but it is not necessary.
+        if (rcl_validate_topic_name(topic.c_str(), &validationResult, &invalidIndex) != RCL_RET_OK)
+        {
+            AZ_Error("ValidateTopic", false, "Call to rcl validation for topic failed");
+            return AZ::Failure(AZStd::string("Unable to validate topic due to rcl error"));
+        }
+
+        if (RCL_TOPIC_NAME_VALID != validationResult)
+        {
+            return AZ::Failure(AZStd::string(rcl_topic_name_validation_result_string(validationResult)));
+        }
+
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateTopicField(void* newValue, [[maybe_unused]] const AZ::Uuid& valueType)
+    {
+        AZStd::string topic(static_cast<const char*>(newValue));
+        return ValidateTopic(topic);
```

Comment:
Seems like this should validate that the TypeId is `azrtti_typeid<const char>`

```
valueType == azrtti_typeid<const char>()
```

Reply:
I am adding a check, but it seems that the actual valueType for this field is AZStd::string. Is this intended? 

Here are the values when called in the good scenario (valueType as called, AZStd::string type UUID, const char type UUID)
{03AAAB3F-5C47-5A66-9EBC-D5FA4DB353C9}, {03AAAB3F-5C47-5A66-9EBC-D5FA4DB353C9}, {3AB0037F-AF8D-48CE-BCA0-A170D18B2C03}

I will commit changes to reflect this, but let me know if this is correct or going to change.

Reply:
Ok if the Uuid is AZStd::string, that means that the void pointer is a pointer to an `AZStd::string`, not a `const char*`, so you will need to update the void* cast here.